src/shared/bpf-object.c

```diff
@@ -0,0 +1,139 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#include "bpf-object.h"
+#include "bpf-program-v2.h"
+#include "fd-util.h"
+#include "set.h"
+
+int bpf_object_new(const unsigned char *mem_buf, size_t size, struct bpf_object **object) {
+        _cleanup_(bpf_object_freep) struct bpf_object *p = NULL;
+        /* Standard error code. */
+        long err;
+
+        assert(mem_buf);
+        assert(size > 0);
+        assert(object);
+
+        DECLARE_LIBBPF_OPTS(
+                        bpf_object_open_opts,
+                        opts);
+
+        p = bpf_object__open_mem(mem_buf, size, &opts);
+        /* Should either point to a valid BPF object or store standard error code. */
+        if (!p)
```

Comment:
Should this check be removed as libbpf doesn't return `NULL` on error but returns the error as a pointer with `ERR_PTR()`? → https://github.com/libbpf/libbpf/blob/051a4009f94d5633a8f734ca4235f0a78ee90469/src/libbpf.c#L7212
The below check with `libbpf_get_error` should be enough, shouldn't it?

Reply:
correct, libbpf_get_error() should be the first to call and check, if it's not an error, it won't be NULL for sure.